soda_machine.rb

```diff
@@ -1,18 +1,27 @@
 class SodaMachine
-  attr_reader :sodas, :cash
+  attr_reader :sodas
+  attr_accessor :cash
 
   def initialize(args = {})
     @sodas = args[:sodas]
     @cash = args[:cash]
   end
 
   def current_inventory_count
+    self.sodas.count
   end
 
   def find_soda(soda_brand)
+    self.sodas.find {|soda| soda.brand == soda_brand}
   end
 
+# return nil unless sold_soda to minimize code
   def sell(soda_brand)
```

Comment:
I like how you structured this method to handle the non-nil result of `find_soda(soda_brand)`.